hyperlink/test/test_url.py

```diff
@@ -231,9 +254,21 @@ def test_roundtrip(self):
         L{URL.to_text} should invert L{URL.from_text}.
         """
         for test in ROUNDTRIP_TESTS:
-            result = URL.from_text(test).to_text()
+            result = URL.from_text(test).to_text(with_password=True)
             self.assertEqual(test, result)
 
+    def test_roundtrip_double_iri(self):
```

Comment:
yeah, i'm down for some hypothesis in the future. i played with it a bit on this PR, but haven't delved deep enough yet.